agent/agent.go

```diff
@@ -241,18 +241,18 @@ func (a *Agent) startCheckTicker() {
 func (a *Agent) startDiscoverTicker() {
 	tick := func() {
 		for _, d := range a.discoveries {
-			var status string
+			var status, result string
 			var err error
 
-			err = d.Discover()
+			result, err = d.Discover()
 			if err != nil {
 				log.Printf("Error while running discovery '%s': %s", d.GetId(), err)
 				status = consulApi.HealthWarning
 			} else {
 				status = consulApi.HealthPassing
 			}
 
-			err = a.consul.Agent().UpdateTTL(d.GetId(), "", status)
+			err = a.consul.Agent().UpdateTTL(d.GetId(), result, status)
```

Comment:
This is the strange thing. If we always return `result` this might not be a correct message in case that the `Discover()` function returns an error.
I suggest the next:
- We store the sent result in `a.consul.Agent()...` depending on which if statement we enter. If we go in the `if` case (error), we will need to unpack the error to get a string. If we are in the `else` side, we directly send the result. You can use the `result` variable to store the error message in the `if` case, that's all. Something like

```
result, err = d.Discover()
if err != nil {
    log.Printf("Error while running discovery '%s': %s", d.GetId(), err)
    result = err.Error()
    status = consulApi.HealthWarning
} else {
...
```